### ISSUE TITLE
Can I use rest.x with projectlombok.org?

### ISSUE THREAD

**Issue by fedotxxl** [restx/restx#185]
Both projects uses annotation processing technique. I'm not familiar with it. I tried `@AllArgsConstructor` annotation on `@Component` - rest.x can't compile it. I think I should specify order of annotations processing - first project lombok, second rest.x

How can I do it in Intellij IDEA / gradle?

**Comment by xhanin** [comment]
I'm not sure how Lombok work and if in proper order it would work. IIUC Lombok uses a non standard technique and modifies the AST of what is compiled during annotation processing, so I don't know if restx annotation processor (which uses standard annotation processing feature and only generates other classes / resources) can be aware of the modified AST.

I don't think I'll have enough time to investigate on that subject, so I'd say that for now integration with Lombok is not supported. If you get it working it would be nice to share though!

**Comment by fedotxxl** [comment]
I use following configuration:

Intellij IDEA: `14.1` + `lombok` plugin
`Compiler` > `Annotation Processors` > enable `Enable annotation processing` + `Obtain processors from project classpath`.

When I ran this configuration I got `Caused by: java.lang.ClassNotFoundException: com.sun.tools.javac.processing.JavacProcessingEnvironment` exception. Solution is described in this thread - http://stackoverflow.com/questions/25607272/compiling-a-java-class-in-memory-with-lombok-annotations-and-java-jdk-8 . I added `compile files("${System.properties['java.home']}/../lib/tools.jar")` and it works! So IDEA compiles restx with lombok.
